Yona/Yona/DaySingleActivityDetail.swift

```diff
@@ -100,33 +110,17 @@ class DaySingleActivityDetail: NSObject {
                 let aCommentLink = link[YonaConstants.jsonKeys.hrefKey] as? String{
                 commentLink = aCommentLink
             }
-            
-        }
-        
-        let range1 =  goalLinks?.range(of: "goals/")?.lowerBound
-        var goal1ID = ""
-        if let txt = goalLinks?.substring(from: range1!) {
-            goal1ID = txt
         }
+    }
+    
+    fileprivate func retrieveGoalDetails(_ allGoals: [Goal]) {
         for goal in allGoals {
-            let range1 =  goal.selfLinks?.range(of: "goals/")?.lowerBound
-            var goal2ID = ""
-            if let txt = goal.selfLinks?.substring(from: range1!) {
-                goal2ID = txt
-            }
-            if goal1ID == goal2ID {
```

Comment:
The above ```if``` condition is the root cause of the issue. Not sure why it was there.
Here they are comparing
goal1ID = Logged-in user goal Link for goal type x.
goal2ID = Buddy user goal link for goal type x. 
This two are will never match.
So this will not set the goal type value when navigating to the no-go alert details of buddy. 
Here, _No Data found error is show to the user forcedly_

Reply:
I agree with your reasoning. Still, I wonder why they did this. Most of the time, people forget to add code rather than writing too much code :)

Anyway, if you have tested the app, the friends timeline as well as the personal timeline, and both are working correctly, then we should go with this.

Reply:
Ok. I'll spend few hours for testing today.

Reply:
What's the outcome of the testing?

Reply:
While testing following scenario covered:
1. Personal timeline dashboard working fine, able to go detail view of each goal.
2. Friends timeline working fine, able to go detail view of each goal. And able start new chat or reply to comment.
3. Notification screen also working fine, able to go inside of personal and friends no go alert.

Reply:
Great. Then we're done with this. I'm going to merge it.